collectors/ebpf.plugin/ebpf_fd.c

```diff
@@ -0,0 +1,533 @@
+                               EBPF_COMMON_DIMENSION_CALL,
+                               NETDATA_APPS_FILE_GROUP,
+                               NETDATA_EBPF_CHART_TYPE_STACKED,
+                               20061,
+                               ebpf_algorithms[NETDATA_EBPF_INCREMENTAL_IDX],
+                               root, NETDATA_EBPF_MODULE_NAME_PROCESS);
+
+    if (em->mode < MODE_ENTRY) {
+        ebpf_create_charts_on_apps(NETDATA_SYSCALL_APPS_FILE_OPEN_ERROR,
+                                   "Fails to open files",
+                                   EBPF_COMMON_DIMENSION_CALL,
+                                   NETDATA_APPS_FILE_GROUP,
+                                   NETDATA_EBPF_CHART_TYPE_STACKED,
+                                   20062,
+                                   ebpf_algorithms[NETDATA_EBPF_INCREMENTAL_IDX],
+                                   root, NETDATA_EBPF_MODULE_NAME_PROCESS);
+    }
+
+    ebpf_create_charts_on_apps(NETDATA_SYSCALL_APPS_FILE_CLOSED,
+                               "Files closed",
+                               EBPF_COMMON_DIMENSION_CALL,
+                               NETDATA_APPS_FILE_GROUP,
+                               NETDATA_EBPF_CHART_TYPE_STACKED,
+                               20063,
+                               ebpf_algorithms[NETDATA_EBPF_INCREMENTAL_IDX],
+                               root, NETDATA_EBPF_MODULE_NAME_PROCESS);
+
+    if (em->mode < MODE_ENTRY) {
+        ebpf_create_charts_on_apps(NETDATA_SYSCALL_APPS_FILE_CLOSE_ERROR,
+                                   "Fails to close files",
+                                   EBPF_COMMON_DIMENSION_CALL,
+                                   NETDATA_APPS_FILE_GROUP,
+                                   NETDATA_EBPF_CHART_TYPE_STACKED,
+                                   20064,
+                                   ebpf_algorithms[NETDATA_EBPF_INCREMENTAL_IDX],
+                                   root, NETDATA_EBPF_MODULE_NAME_PROCESS);
+    }
+}
+
+/**
+ * Create global charts
+ *
+ * Call ebpf_create_chart to create the charts for the collector.
+ *
+ * @param em a pointer to the structure with the default values.
+ */
+static void ebpf_create_fd_global_charts(ebpf_module_t *em)
+{
+    ebpf_create_chart(NETDATA_FILESYSTEM_FAMILY,
+                      NETDATA_FILE_OPEN_CLOSE_COUNT,
+                      "Open and close calls",
+                      EBPF_COMMON_DIMENSION_CALL,
+                      NETDATA_FILE_GROUP,
+                      NULL,
+                      NETDATA_EBPF_CHART_TYPE_LINE,
+                      NETDATA_CHART_PRIO_EBPF_FD_CHARTS,
+                      ebpf_create_global_dimension,
+                      fd_publish_aggregated,
+                      NETDATA_FD_SYSCALL_END,
+                      NETDATA_EBPF_MODULE_NAME_FD);
+
+    if (em->mode < MODE_ENTRY) {
+        ebpf_create_chart(NETDATA_FILESYSTEM_FAMILY,
+                          NETDATA_FILE_OPEN_ERR_COUNT,
```

Comment:
I don't see any charts with open/close errors neither aggregated nor in the apps section.

Reply:
I fixed this with last commit.